src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,106 @@
 package core.basesyntax;
 
+import java.util.NoSuchElementException;
+
 public class ArrayList<T> implements List<T> {
+    private static final int DEFAULT_CAPACITY = 10;
+    private static final double GROWTH_FACTOR = 1.5;
+    private Object[] elementData;
```

Comment:
Why you didn't use T[] element data? Why is it Object[]?

Reply:
The default implementation of AarayList uses an array of objects as an item store. Perhaps it can be explained by the fact that we don't work directly with this storage, and it makes no sense to use generics there, but I could be wrong.